editor/plugins/game_view_plugin.cpp

```diff
@@ -286,11 +397,43 @@ void GameView::_notification(int p_what) {
 			select_mode_button[RuntimeNodeSelect::SELECT_MODE_SINGLE]->set_button_icon(get_editor_theme_icon(SNAME("ToolSelect")));
 			select_mode_button[RuntimeNodeSelect::SELECT_MODE_LIST]->set_button_icon(get_editor_theme_icon(SNAME("ListSelect")));
 
+			hide_selection->set_button_icon(get_editor_theme_icon(hide_selection->is_pressed() ? SNAME("GuiVisibilityHidden") : SNAME("GuiVisibilityVisible")));
+			embedded_button->set_button_icon(get_editor_theme_icon(SNAME("EmbeddedProcess")));
+			auto_focus_button->set_button_icon(get_editor_theme_icon(SNAME("AutoFocus")));
+			keep_aspect_button->set_button_icon(get_editor_theme_icon(SNAME("KeepAspect")));
+
 			hide_selection->set_button_icon(get_editor_theme_icon(hide_selection->is_pressed() ? SNAME("GuiVisibilityHidden") : SNAME("GuiVisibilityVisible")));
 
 			camera_override_button->set_button_icon(get_editor_theme_icon(SNAME("Camera")));
 			camera_override_menu->set_button_icon(get_editor_theme_icon(SNAME("GuiTabMenuHl")));
 		} break;
+
+		case NOTIFICATION_READY: {
+			if (DisplayServer::get_singleton()->has_feature(DisplayServer::FEATURE_WINDOW_EMBEDDING)) {
+				// Embedding available.
+				embedded_button->set_pressed(EditorSettings::get_singleton()->get_project_metadata("game_view", "embedded", true));
+				auto_focus_button->set_pressed(EditorSettings::get_singleton()->get_project_metadata("game_view", "auto_focus", true));
+				keep_aspect_button->set_pressed(EditorSettings::get_singleton()->get_project_metadata("game_view", "keep_aspect", true));
```

Comment:
Why are those using the project metadata instead of the plugin state?

Reply:
I tested it, I did not known Plugin States existed! The problem I see is that plugin states are associated with an open scene and are saved when a scene is saved/closed (check: `EditorData::save_edited_scene_state`). I don't think these settings should be saved by scene, they are more global settings can I believe they should be "by project" and that's why I putted in project metadata. What do you think?

Reply:
That's actually a great point. Perhaps the other settings should be converted to use the project metadata as well then.